File: BACK-END/KuentecoApp/src/main/java/org/kuenteco/backend/dto/logic/transaction/bancolombia/response/TokenResponseDTO.java
package org.kuenteco.backend.dto.logic.transaction.bancolombia.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

// DTO para token de autenticación
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TokenResponseDTO {
    @JsonProperty("access_token")
    private String accessToken;

    @JsonProperty("token_type")
    private String tokenType;

    @JsonProperty("expires_in")
    private Long expiresIn;

    private String scope;

    @JsonProperty("refresh_token")
    private String refreshToken;
}

File: FRONT-END/lib/dto/app/transaction/bancolombia/response/token_response_dto.dart
class TokenResponseDTO {
  final String accessToken;
  final String tokenType;
  final int expiresIn;
  final String scope;
  final String refreshToken;

  TokenResponseDTO({
    required this.accessToken,
    required this.tokenType,
    required this.expiresIn,
    required this.scope,
    required this.refreshToken,
  });

  factory TokenResponseDTO.fromJson(Map<String, dynamic> json) {
    return TokenResponseDTO(
      accessToken: json['access_token'],
      tokenType: json['token_type'],
      expiresIn: json['expires_in'],
      scope: json['scope'] ?? '',
      refreshToken: json['refresh_token'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'access_token': accessToken,
      'token_type': tokenType,
      'expires_in': expiresIn,
      'scope': scope,
      'refresh_token': refreshToken,
    };
  }
}
